### PR TITLE
pom.xml files - refactoring

### PR DESCRIPTION
**Pull request by svenruppert**
I just removed redundant dependency definitions, centralized the version definitions from 
dependencies and modules. added profile with the maven plugin **versions** to support
version management (checking for new versions)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,118 +1,274 @@
...
+    <module>fn-spring-cloud-function</module>
+    <module>examples</module>
+  </modules>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+
+    <!--TDD-->
+    <!--<jacoco.version>0.7.9</jacoco.version>-->
```
Comment (mjg123) [inline pom.xml]: I guess we can remove this?

File: pom.xml [hunk 2]
```diff
@@ -1,118 +1,274 @@
...
+    <junit.version>4.12</junit.version>
+    <mockito.version>2.8.47</mockito.version>
+    <assertj-core.version>3.6.2</assertj-core.version>
+    <pitest.version>1.4.0</pitest.version>
+
+    <slf4j.version>1.7.25</slf4j.version>
+
+    <commons-io.version>2.5</commons-io.version>
+    <jackson.version>2.9.6</jackson.version>
+    <httpcore.version>4.4.6</httpcore.version>
```
Comment (mjg123) [inline pom.xml]: Several of these dependency pins are not latest versions any more.  What do you think about updating them as part of this PR?

File: pom.xml [hunk 3]
```diff
@@ -1,118 +1,274 @@
...
+        <artifactId>dockerfile-maven-extension</artifactId>
+        <version>1.3.1</version>
+      </extension>
+    </extensions>
+  </build>
+
+
+  <profiles>
+    <profile>
+      <id>_qm-qs</id>
```
Comment (mjg123) [inline pom.xml]: Hmmmm what is this profile for?

File: testing-junit4/pom.xml [hunk 4]
```diff
@@ -15,27 +15,17 @@
...
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
-      <version>${junit.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>org.assertj</groupId>
-      <artifactId>assertj-core</artifactId>
-      <version>${assertj-core.version}</version>
-      <scope>test</scope>
+      <scope>compile</scope>
```
Comment (mjg123) [inline testing-junit4/pom.xml]: This is the default scope anyway, but makes sense to be clear that we are (unusually) adding junit as a real dependency, perhaps?

**Comment by svenruppert** [comment]
will update regarding your comments...

**Pull request merged**